packages/server/src/fhir/accesspolicy.ts

```diff
@@ -113,3 +118,59 @@ async function buildAccessPolicyResources(
   }
   return JSON.parse(json) as AccessPolicy;
 }
+
+/**
+ * Updates the access policy to include project admin rules.
+ * This includes ensuring no admin rights for non-admins and restricted access for admins.
+ * @param login The user login.
+ * @param membership The active project membership.
+ * @param accessPolicy The existing access policy.
+ * @returns Updated access policy with all project admin rules applied.
+ */
+function applyProjectAdminAccessPolicy(
+  login: Login,
+  membership: ProjectMembership,
+  accessPolicy: AccessPolicy | undefined
+): AccessPolicy | undefined {
+  if (login.superAdmin) {
+    // If the user is a super admin, then do not apply any additional access policy rules.
+    return accessPolicy;
+  }
+
+  if (!membership.admin && !accessPolicy) {
+    // Not a project admin and no access policy, so return default access.
+    return undefined;
+  }
+
+  if (accessPolicy) {
+    // If there is an existing access policy
+    // Remove any references to project admin resource types
+    accessPolicy.resource = accessPolicy.resource?.filter(
+      (r) => r.resourceType !== 'Project' && r.resourceType !== 'ProjectMembership'
```

Comment:
It would be safer to export the constant `projectAdminResourceTypes` from repo and and refer to them here, in case we expand the contents of `projectAdminResourceTypes` in the future

Reply:
👍